lib/logstash/outputs/elasticsearch/http_client/manticore_adapter.rb

```diff
@@ -66,23 +66,54 @@ def perform_request(url, method, path, params={}, body=nil)
 
       request_uri = format_url(url, path)
       request_uri_as_string = remove_double_escaping(request_uri.to_s)
-      resp = @manticore.send(method.downcase, request_uri_as_string, params)
-
-      # Manticore returns lazy responses by default
-      # We want to block for our usage, this will wait for the repsonse
-      # to finish
-      resp.call
+      begin
+        resp = @manticore.send(method.downcase, request_uri_as_string, params)
+        # Manticore returns lazy responses by default
+        # We want to block for our usage, this will wait for the response to finish
+        resp.call
+      rescue ::Manticore::ManticoreException => e
+        handle_request_error(e, request_uri_as_string)
```

Comment:
The method doesn't handle the exception, it logs the exception and re-raise. So I would split the two tasks and make evident that an exception is raised, something like:
```ruby
begin
  resp = @manticore.send(method.downcase, request_uri_as_string, params)
  resp.call
rescue ::Manticore::ManticoreException => e
  log_error(e)
  raise ::LogStash::Outputs::ElasticSearch::HttpClient::Pool::HostUnreachableError.new(e, url)
end  
```

where `log_error` is:
```ruby
def log_error(e)
  # here goes the code of #handle_request_error without the raise clause
end
```

In this way it makes evident that the `rescue` does always a re-throw of the exception, and cleanly separate the responsibilities

Reply:
thanks - makes sense. updated.